public/js/dataitem.js

```diff
@@ -131,30 +166,19 @@ $("main").on('click', '#add_dataitem_button_submit', function () {
         isValid = false;
         msgBanner.error('Linkback must be in valid URL format');
     }
+    var file = $('#addImage')[0].files[0];
+    if (file && file.size > 4 * 1024 * 1024) { 
+        isValid = false;
+        msgBanner.error('The image size should be less than 4MB');
+    }
```

Comment:
Please make the max size (MAX_UPLOAD_IMAGE_SIZE) as a configuration in .env and default to 4MB.

Reply:
Also add the server validation for the size.